pkg/schemas/generator.go

```diff
@@ -105,11 +137,125 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 			documents[documentName] = document
 		}
 		document.Definitions[context.definitionNameFor(documentName, typeIdent)] = typeSchema
+
+		// Generate a schema for types with "fybrik:validation:object" marker
+		info, knownInfo := parser.Types[typeIdent]
+		if knownInfo {
+			if info.Markers.Get(objectMarker.Name) != nil {
+				documentName := fmt.Sprintf("%s.json", info.Name)
+				document, exists := documents[documentName]
+				listFields, _ := context.getFields(typeIdent)
+				schemaPtr := parser.Schemata[typeIdent]
+				context.removeExtraProps(typeIdent, &schemaPtr, &listFields)
+				if !exists {
+					document = schemaPtr.DeepCopy()
+					document.Title = documentName
+					document.Definitions = make(apiext.JSONSchemaDefinitions)
+					documents[documentName] = document
+				}
+
+				for _, fieldType := range listFields {
+					typeSchemaField := parser.Schemata[fieldType]
+					context.removeExtraProps(fieldType, &typeSchemaField, &listFields)
+					document.Definitions[context.definitionNameFor(documentName, fieldType)] = typeSchemaField
+				}
+			}
+		}
 	}
 
 	return g.output(documents)
 }
 
+// Get the fields that related to taxonomy (has a taxonomy child)
+// It returns true iff the type has a taxonomy child
+func (context *GeneratorContext) getFields(typ crd.TypeIdent) ([]crd.TypeIdent, bool) {
+	ListFields := []crd.TypeIdent{}
+	isTaxonomy := false
+	info, knownInfo := context.parser.Types[typ]
+	if !knownInfo {
+		return ListFields, false
+	} else {
+		fields := info.Fields
+		for _, field := range fields {
+			typeName := field.RawField.Type
+			typeNameStr := fmt.Sprintf("%s", typeName)
+			if strings.Contains(typeNameStr, "taxonomy") {
+				isTaxonomy = true
+				continue
+			}
```

Comment:
Consider doing a more exact match then just Contains. Also consider making the package to filter by a marker parameter instead of treating taxonomy as a magic word.

Reply:
Done.